src/internal/connector/onedrive/restore.go

```diff
@@ -65,18 +68,20 @@ func RestoreCollections(
 
 	// Iterate through the data collections and restore the contents of each
 	for _, dc := range dcs {
-		parentPerms, ok := parentPermissions[dc.FullPath().String()]
-		if !ok {
-			onedrivePath, err := path.ToOneDrivePath(dc.FullPath())
-			if err != nil {
-				errUpdater(dc.FullPath().String(), fmt.Errorf("invalid restore path"))
-			}
+		if opts.RestorePermissions {
+			parentPerms, ok = parentPermissions[dc.FullPath().String()]
+			if !ok {
+				onedrivePath, err := path.ToOneDrivePath(dc.FullPath())
+				if err != nil {
+					errUpdater(dc.FullPath().String(), fmt.Errorf("invalid restore path"))
+				}
 
-			if len(onedrivePath.Folders) == 0 {
-				// root directory will not have permissions
-				parentPerms = []UserPermission{}
-			} else {
-				errUpdater(dc.FullPath().String(), fmt.Errorf("unable to find parent permissions"))
+				if len(onedrivePath.Folders) == 0 {
+					// root directory will not have permissions
+					parentPerms = []UserPermission{}
+				} else {
+					errUpdater(dc.FullPath().String(), fmt.Errorf("unable to find parent permissions"))
+				}
```

Comment:
I know I'm picking at old code, but I'm a bit confused here, so I have to ask about it.  So, if we have a non-zero list of folders, then we record an "unable to find" error?  That seems counter-intuitive.

Reply:
We try to get premissions for the parent folder [here](https://github.com/alcionai/corso/blob/8c91de1ca2be61145cd7c73cd66f62c8fae5642f/src/internal/connector/onedrive/restore.go#L72). Failing that we check if the parent is the root, which might not have separate permissions associated to it. If permissions are not available form `parentPermissions` and it is not the root, then we fail saying that we are not able to find permissions of the parent directory.

Reply:
Hmm, I think a message like `"directory has no permissions"` is more accurate in this case.

Reply:
Just to clarify, we do have permissions for the folder that we are about to restore, but we are not able to find the permissions for its parent(which should ideally have been restored already) and due to that we are not able to figure out the final permissions for this folder as it has to be computed based on what the parent has. Let me change the error message to "unable to compute item permissions", but let me know if you still think it should be something else.